### PR TITLE
feat: add wishlist-page

### LINKED ISSUE #17: add wish-list page
here I added wishlist

### PR DESCRIPTION
**Pull request by Bhakti199**
Here I added wishlist page #17 . on clicking the heart icon the item is moved to wishlist. You can item add to wishlist from product-listing page and cart-page also and vice-versa for remove item from wishlist.

Screenshot (1330)

### CODE REVIEW

File: src/Pages/WishListPage/WishListPage.jsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                        }
+                      >
+                        Add to cart
+                      </button>
+                    </div>
+                  </div>
+                </section>
+              ))
+            : ""}
+        </div>
```
Comment (JantuDeb) [inline src/Pages/WishListPage/WishListPage.jsx]: You can redirect user to product listing page if no item present in wishlist page

File: src/Context/FilterContext/WishListUtils.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export const addToWishList = (itemToAdd, wishList) => {
+  if (wishList.length === 0) return [itemToAdd];
+  return wishList.some((item) => item._id === itemToAdd._id)
+    ? wishList
+    : [...wishList, itemToAdd];
+};
```
Comment (JantuDeb) [inline src/Context/FilterContext/WishListUtils.js:7]: This is a good use case of `some()` method

**Review by JantuDeb — APPROVED** [review]
Added a comment

File: src/App.js [hunk 3]
```diff
@@ -13,6 +18,7 @@ function App() {
         <Route path="/mockman" element={<MockMan />} />
         <Route path="/product-listing" element={<ProductListingPage />} />
         <Route path="/cart-page" element={<CartPage />} />
+        <Route path="/wishlist-page" element={<WishListPage />} />
       </Routes>
```
Comment (dikshadhote) [inline src/App.js:22]: Can create separate file for route and then import in App

**Review by dikshadhote — APPROVED** [review]
The code is very well structured.
Good work !

File: src/Pages/WishListPage/WishListPage.jsx (hunk 1 above)
Comment (Bhakti199) [reply src/Pages/WishListPage/WishListPage.jsx]: okay

**Pull request merged**